appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/palette/YoungAndroidPalettePanel.java

```diff
@@ -75,6 +85,39 @@ public YoungAndroidPalettePanel(YaFormEditor editor) {
     categoryPanels = new HashMap<ComponentCategory, VerticalPanel>();
     simplePaletteItems = new HashMap<String, SimplePaletteItem>();
 
+    componentTrie = new Trie();
+    translationMap = new HashMap();
+    final VerticalPanel panel = new VerticalPanel();
+    panel.setWidth("100%");
+
+    //Load Component strings to Trie
+    for (String component : COMPONENT_DATABASE.getComponentNames()) {
+      String translationName = ComponentsTranslation.getComponentName(component).toLowerCase();
+      componentTrie.insert(translationName);
+      translationMap.put(translationName, component);
+    }
+
+    // initialize the panel that holds search
+   
+    final TextBox searchText = new TextBox();
+    Button searchButton = new Button(MESSAGES.searchComponentButton());
+    Button clearButton = new Button(MESSAGES.clearComponentsButton());
+
+    // buttonPanel holds search and clear button
+    final HorizontalPanel buttonPanel = new HorizontalPanel();
+
+    buttonPanel.add(searchButton);
+    buttonPanel.add(clearButton);
+    panel.add(searchText);
+    panel.add(buttonPanel);
+    
+    buttonPanel.setSpacing(2);
+    
+    stackPalette.setWidth("100%");
+    initWidget(stackPalette);
+
+    stackPalette.add(panel, "Search for Component");
```

Comment:
Rather than add the search panel as a part of the stack panel, I would prefer if you add the `stackPanel` to the `panel` and then call `initWidget(panel)` instead of `initWidget(stackPanel)`.